src/com/spikes2212/genericsubsystems/drivetrains/commands/DriveTankWithPID.java

```diff
@@ -2,332 +2,318 @@
 
 import java.util.function.Supplier;
 
-import com.spikes2212.genericsubsystems.LimitedSubsystem;
-import com.spikes2212.genericsubsystems.commands.MoveLimitedSubsystemWithPID;
 import com.spikes2212.genericsubsystems.drivetrains.TankDrivetrain;
+import com.spikes2212.utils.PIDSettings;
 
 import edu.wpi.first.wpilibj.PIDController;
 import edu.wpi.first.wpilibj.PIDSource;
 import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj.command.Command;
-import edu.wpi.first.wpilibj.command.PIDCommand;
+
+/**
+ * This command moves a {@link TankDrivetrain} using wpilib's
+ * {@link PIDController}. It also waits a specified amount of time after the
+ * error is within the given tolerance before stopping the PID Loop to make sure
+ * the drivetrain doesn't go past and remain beyond the setpoint.
+ *
+ * @author Omri "Riki" Cohen
+ * @see TankDrivetrain
+ * @see PIDController
+ * @see PIDSettings
+ */
 
 public class DriveTankWithPID extends Command {
 
-    private TankDrivetrain tankDrivetrain;
-    private double KP;
-    private double KI;
-    private double KD;
-    private double tolerance;
-    private Supplier<Double> leftSetpoint;
-    private Supplier<Double> rightSetpoint;
-    private PIDSource leftSource;
-    private PIDSource rightSource;
-    private PIDController leftMovmentControl;
-    private PIDController rightMovmentControl;
-    private double lastTimeNotOnTarget;
-    private double waitTime = 0.5;
+	private TankDrivetrain tankDrivetrain;
+	private Supplier<Double> leftSetpoint;
+	private Supplier<Double> rightSetpoint;
+	private PIDSettings PIDSettings;
+	private PIDSource leftSource;
+	private PIDSource rightSource;
+	private PIDController leftMovmentControl;
+	private PIDController rightMovmentControl;
+	private double lastTimeNotOnTarget;
+
+	/**
+	 * Gets the Proportional coefficient of the PID loop of this command.
+	 *
+	 * @return the current Proportional coefficient.
+	 * @see PIDController#getP()
+	 * @see PIDSettings#getKP()
+	 */
 
-    /**
-     * Sets the Proportional coefficient of the PID loop of this command.
-     *
-     * @param P the new Proportional coefficient.
-     * @see PIDController#setPID(double, double, double)
-     */
-    public void setP(double P) {
-        KP = P;
-    }
+	public double getP() {
+		return PIDSettings.getKP();
+	}
```

Comment:
call these getKP and getKI and so on, because that's the way you called them in PIDSettings

Reply:
Wouldn't it be better to have getPIDSettings?